opal/runtime/opal_params_core.h

```diff
@@ -59,6 +59,7 @@ OPAL_DECLSPEC extern int opal_initialized;
 OPAL_DECLSPEC extern bool opal_built_with_cuda_support;
 OPAL_DECLSPEC extern bool opal_built_with_rocm_support;
 OPAL_DECLSPEC extern bool opal_built_with_ze_support;
+OPAL_DECLSPEC extern bool opal_async_progress_thread_spawned;
```

Comment:
This patch removes the `OPAL_ENABLE_PROGRESS_THREADS` pre-processor guard from many places (like https://github.com/open-mpi/ompi/pull/13088/files#diff-b23aacc904fb4f13b003d1c716bf6db21c5c490a319b58a3ab70e8f3a8226885L48). Can we define `opal_async_progress_thread_spawned` to `false` in the header if progress threads are explicitly disabled at compile time? That makes the code cleaner and allows the compiler to optimize away code that is explicitly disabled.

Reply:
Yes, done in new updated version. Thanks for proposal.